src/lib/deterministic-fsm/deterministic-transition.fn.ts

```diff
@@ -0,0 +1,20 @@
+import { MachineEvent, MachineTransitionFn, MachineState, MachineNode } from '../state-machine-api';
+
+export type DeterministicTransitions = Map<MachineNode, Map<MachineEvent, MachineNode>>;
+
+export const deterministicTransitionFn: (transitions: DeterministicTransitions) => MachineTransitionFn<void> = (
+  transitions: DeterministicTransitions,
+) => (currentState: MachineState<void>, $event: MachineEvent): MachineState<void> => {
```

Comment:
why event has the prefix $? its a convention?

Reply:
event is a keyword in js